src/util/logging.cpp

```diff
@@ -241,4 +241,31 @@ void Logging::shutdown() {
     }
 }
 
+namespace {
+
+inline QString preambleString(const char* logContext) {
+    if ((logContext == nullptr) || (strlen(logContext) == 0)) {
+        return QString();
+    } else {
+        return QString("%1 -").arg(logContext);
+    }
+}
+
+inline QString preambleString(const QString& logContext) {
+    if (logContext.isEmpty()) {
+        return QString();
+    } else {
+        return QString("%1 -").arg(logContext);
+    }
+}
+
+} // anonymous namespace
+
+Logger::Logger(const char* logContext)
+    : m_preambleChars(preambleString(logContext).toLocal8Bit()) {
```

Comment:
Can't we return the local 8 bit QByteArray from "preambleString()"? 
I think we can get around the intermediate QString by just append QByteArrays.